load-tests/src/load-test.ts

```diff
@@ -0,0 +1,273 @@
+  scenarios: {
+    insert_plans: {
+      // This defines the function that this scenario will run
+      exec: 'insert_plans',
+
+      // "VUs" (virtual users) are threads that run the function we referenced above in a loop, simulating real users.
+      // Pre-allocating VUs simply saves on run-time resources
+      preAllocatedVUs: 50,
+
+      // When the executor is `ramping-arrival-rate`, the stages define the number of requests
+      // per second we want to target. The k6 scheduler will then automatically scale the number of VUs
+      // in order to hit that desired rate
+
+      // https://k6.io/docs/using-k6/scenarios/executors/
+      executor: 'ramping-arrival-rate',
+      stages: [
+        // Start 10 iterations per second for the first 30 seconds,
+        { target: 10, duration: '30s' },
+        // Then, linearly scale up to 100 iterations/second for a minute,
+        { target: 50, duration: '1m' },
+        // Then scale down again over 30 seconds.
+        { target: 10, duration: '30s' },
+      ],
+    },
+
+    insert_random_activities: {
+      exec: 'insert_random_activities',
+      executor: 'ramping-arrival-rate',
+      // This scenario will only start after 10 seconds, allowing plans to populate first
+      startTime: "10s",
+      preAllocatedVUs: 50,
+      stages: [
+        { target: 10, duration: '30s' },
+        { target: 50, duration: '1m' },
+        { target: 10, duration: '30s' },
+      ],
+    },
+
+    insert_targeted_activities: {
+      exec: 'insert_targeted_activities',
+      // Here we use a different executor, which spawns a set number of VUs, each running a set number of iterations
+      executor: 'constant-vus',
+      startTime: '5s',
+      vus: 50,
+      duration: '30s'
+    },
+
+    run_targeted_simulations: {
+      exec: 'run_targeted_simulations',
+      startTime: '1m',
+      // Another different executor, which runs keeps VUs constant, running as many iterations as possible
+      executor: 'constant-vus',
+      vus: 2,
+      duration: "30s"
+    },
+
+    run_random_simulations: {
+      exec: 'run_random_simulations',
+      startTime: '10s',
+      executor: 'ramping-arrival-rate',
+      preAllocatedVUs: 10,
+      stages: [
+        { target: 5, duration: '30s' },
+        { target: 10, duration: '1m' },
+        { target: 5, duration: '30s' },
+      ],
+    },
+
+    get_effective_args: {
+      exec: 'get_effective_args',
+      executor: 'ramping-arrival-rate',
+      preAllocatedVUs: 10,
+      stages: [
+        { target: 5, duration: '30s' },
+        { target: 10, duration: '1m' },
+        { target: 5, duration: '30s' },
+      ],
+    },
+  },
+
+  // If the statistics of the load test fall below the following threshold, it with exit with a nonzero exit code
+  thresholds: {
+    http_req_failed: ['rate<0.01'], // http errors should be less than 1%
+    http_req_duration: ['p(50)<200', 'p(95)<1000'], // 90% below 400ms, 95% of requests should be below 1000ms
```

Comment:
Minor, non-blocking: Clarity:
```suggestion
    http_req_duration: ['p(90)<400', 'p(95)<1000'], // 90% of requests should be below 400ms, 95% below 1000ms
```

Although you force-pushed changed this threshold so that half our queries need to be below 200ms. Can you clarify why this was necessary?

Reply:
Just trial and error finding a decent load for our github actions. These numbers can (and probably should) be changed when running heavier tests locally, but are intentionally lax for these automated tests.

Reply:
> Just trial and error finding a decent load for our github actions. 

Do we get slower responses in GH actions?

> These numbers can (and probably should) be changed when running heavier tests locally, but are intentionally lax for these automated tests.

Is this statement in the upcoming README? If not, then either there or above where these thresholds are in the code would be a good place to document this expectation.